bugzilla/base.py

```diff
@@ -1131,20 +1134,52 @@ def getbug(self, objid,
             extra_fields=extra_fields)
         return Bug(self, dict=data, autorefresh=self.bug_autorefresh)
 
+    def _to_bug(self, bug_data):
+        if bug_data:
+            return Bug(self, dict=bug_data, autorefresh=self.bug_autorefresh)
+        return None
+
     def getbugs(self, idlist,
                 include_fields=None, exclude_fields=None, extra_fields=None,
-                permissive=True):
+                permissive=True, limit=None):
         """
         Return a list of Bug objects with the full complement of bug data
         already loaded. If there's a problem getting the data for a given id,
         the corresponding item in the returned list will be None.
         """
         data = self._getbugs(idlist, include_fields=include_fields,
             exclude_fields=exclude_fields, extra_fields=extra_fields,
-            permissive=permissive)
-        return [(b and Bug(self, dict=b,
-                           autorefresh=self.bug_autorefresh)) or None
-                for b in data]
+            permissive=permissive, limit=limit)
+        return [self._to_bug(b) for b in data]
```

Comment:
When I look at the git history, I see that somebody added the construct `(b and Bug(...)) or None` deliberately. But from personal usage experience, I do not recall ever seeing a `None` instead of a `Bug` instance.

So, I would say, this simplifications makes a lot of sense :+1: 

@crobinso Would you agree?

Reply:
I believe old bugzilla instances would return ` None` if `'permissive=True`. Maybe we don't need to maintain it anymore. Safest thing to do is keep it intact unless someone wants to dig through old bugzilla docs.